app/api/logging/audit.py

```diff
@@ -19,54 +19,93 @@
 logging.addLevelName(AUDIT, "AUDIT")
 
 
-def init_security_logging() -> None:
-    sys.addaudithook(audit_hook)
+def init() -> None:
+    """Initialize the audit logging module to start
+    logging security audit events."""
+    sys.addaudithook(handle_audit_event)
+
+
+def handle_audit_event(event_name: str, args: tuple[Any, ...]) -> None:
+    # Define events to log and the arguments to log for each event.
+    # For more information about these events and what they mean, see https://peps.python.org/pep-0578/#suggested-audit-hook-locations
+    # For the full list of auditable events, see https://docs.python.org/3/library/audit_events.html
+    # Define this variable locally so it can't be modified by other modules.
+    #
+    # Events from the suggested audit hook locations in PEP 578 that we aren't logging:
+    #
+    # compile and import
+    #     Detects when code is being compiled or imported.
+    #
+    #     Why we aren't logging:
+    #     Logging as part of regular imports is too noisy.
+    #
+    # sys._getframe
+    #     Detects when code is accessing frames directly.
+    #
+    #     Why we aren't logging:
+    #     The logging module itself calls getFrame
+    #     (see https://github.com/python/cpython/blob/bd7903967cd2a19ebc842dd1cce75f60a18aef02/Lib/logging/__init__.py#L170)
+    #     so logging on this event would trigger an infinite loop unless
+    #     logging._srcfile is set to None
+    #     See https://github.com/python/cpython/blob/bd7903967cd2a19ebc842dd1cce75f60a18aef02/Lib/logging/__init__.py#L179-L189
+    #
+    # object.__getattr__
+    #     Detect access to restricted attributes. This event is raised for any built-in members
+    #     that are marked as restricted, and members that may allow bypassing imports.
+    #
+    #     Why we aren't logging:
+    #     The logging module itself accesses frame.f_code.co_filename
+    #     (see https://github.com/python/cpython/blob/bd7903967cd2a19ebc842dd1cce75f60a18aef02/Lib/logging/__init__.py#L202)
+    #     which calls object.__getattr__ under the hood
+    #     (see https://github.com/python/cpython/blob/main/Objects/frameobject.c#L89-L95)
+    #
+
+    EVENTS_TO_LOG = {
+        # Detect dynamic execution of code objects. This only occurs for explicit
+        # calls, and is not raised for normal function invocation.
+        "exec": ("code_object",),
+        # Detect when a file is about to be opened. path and mode are the usual
+        # parameters to open if available, while flags is provided instead of
+        # mode in some cases.
+        "open": ("path", "mode", "flags"),
```

Comment:
Can we add a few of the `os` ones that seem important:

```suggestion
        "open": ("path", "mode", "flags"),
        # Detect when a signal is sent to a process.
        "os.kill": ("pid", "sig"),
        # Detect when a file is deleted.
        "os.remove": ("path", "dir_fd"),
        # Detect when a file is renamed.
        "os.rename": ("src", "dst", "src_dir_fd", "dst_dir_fd"),
```

Reply:
when i tried logging os.remove, it really messes up the unit tests (causes them to go into some sort of infinite loop, since i think every test calls os.remove to close the root logger's streamhandler sys.stdout or something. i'm not really sure what's going on but it's a bit of a rabbit hole so i'm going to put this into the backlog as a future todo. sorry about that.